src/plugins/data/server/ui_settings.ts

```diff
@@ -668,6 +668,17 @@ export function getUiSettings(): Record<string, UiSettingsParams<unknown>> {
       }),
       schema: schema.string(),
     },
+    [UI_SETTINGS.DATASOURCE_PLACEHOLDER]: {
```

Comment:
can we put this ui_settings into a data_source management plugin?

Reply:
actually, we may want to have a `data_source` plugin to define the data source saved object type, then another `data_source_management` plugin for the management UI